hash_practice/exercises.py

```diff
@@ -2,28 +2,101 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagrams_table = {}
+
+    # iterate over each word: sort it, then check if it's in anagrams table.
+    # if not, add it
+    # -> creates a table wherein each key is the sorted anagram and its value is an array of anagram strings 
+    for word in strings:
+        sorted_word = "".join(sorted(word))
+        if sorted_word in anagrams_table:
+            anagrams_table[sorted_word].append(word)
+        else:
+            anagrams_table[sorted_word] = [word]
+
+    # the anagrams table already has the anagrams grouped - just need to
+    # get each array and add to a list
+    # I am lazy and using this here built-in method
+    return list(anagrams_table.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 Really nicely done, you didn't even have to sort!  See below for a note on time/space.